File: board-admin-backend/src/main/java/com/portSrilanka/board_admin_backend/dto/comment/CommentRequest.java
package com.portSrilanka.board_admin_backend.dto.comment;

import lombok.Data;

@Data
public class CommentRequest {
    private Long meetingId;
    private Long paperId;
    private String commentText;
    private boolean annotated;
    private String visibility;
    private Integer pageNumber;
    private java.util.Set<Long> selectedUserIds;
}

File: board-admin-frontend/frontend/lib/features/comments/model/comment_request.dart
class CommentRequest {
  final int? meetingId;
  final int? paperId;
  final String commentText;
  final bool annotated;
  final String visibility;
  final int? pageNumber;
  final List<int> selectedUserIds;

  CommentRequest({
    this.meetingId,
    this.paperId,
    required this.commentText,
    required this.annotated,
    this.visibility = 'ALL_PARTICIPANTS',
    this.pageNumber,
    this.selectedUserIds = const [],
  });

  Map<String, dynamic> toJson() {
    return {
      'meetingId': meetingId,
      'paperId': paperId,
      'commentText': commentText,
      'annotated': annotated,
      'visibility': visibility,
      'pageNumber': pageNumber,
      'selectedUserIds': selectedUserIds,
    };
  }
}
